client/renderSDL/SDL_Extensions.cpp

```diff
@@ -543,28 +543,42 @@ void CSDL_Ext::putPixelWithoutRefreshIfInSurf(SDL_Surface *ekran, const int & x,
 }
 
 template<int bpp>
-void CSDL_Ext::convertToGrayscaleBpp(SDL_Surface * surf, const Rect & rect )
+void loopOverPixel(SDL_Surface * surf, const Rect & rect, std::function<void(int &r, int &g, int &b)> func)
```

Comment:
```suggestion
void loopOverPixel(SDL_Surface * surf, const Rect & rect, Functor func)
```
Since this is template anyway, then it would be better to provide function not as std::function, but as template parameter, to avoid forced function call for every pixel